app/src/org/odk/collect/android/widgets/QuestionWidget.java

```diff
@@ -352,7 +353,27 @@ private void stripUnderlines(TextView textView) {
             s.setSpan(span, start, end, 0);
         }
         textView.setText(s);
-    }    
+    }
+
+    public void setQuestionText(TextView textView, FormEntryPrompt prompt){
+        if(prompt.getMarkdownText() != null){
+            textView.setText(forceMarkdown(prompt.getMarkdownText()));
+            textView.setMovementMethod(LinkMovementMethod.getInstance());
+            // Wrap to the size of the parent view
+            textView.setHorizontallyScrolling(false);
+        } else {
+            textView.setText(mPrompt.getLongText());
+        }
+    }
+
+    public void setChoiceText(TextView choiceText, FormEntryPrompt prompt, int index){
+        String markdownText = mPrompt.getSelectItemMarkdownText(mPrompt.getSelectChoices().get(index));
```

Comment:
getSelectChoices here can involve a few callouts and should be considered an expensive method. I'd save this result inline to prevent it from being re-fired on like 374. Good otherwise.